### PR TITLE
feat: add test resource configuration and CLI flags

### PR DESCRIPTION
**Pull request by javacruft**
## Summary

This PR adds support for specifying test-specific resource requirements in melange configuration files and via CLI flags. This enables external schedulers (like elastic build) to provision appropriately-sized test environments that may differ from build environments.

### Configuration Schema
- **New field**: `package.test-resources` with identical structure to `package.resources`
  - `cpu`: CPU count (e.g., `"4"`, `"8"`)
  - `cpumodel`: CPU model requirement (e.g., `"intel-xeon"`)
  - `memory`: Memory allocation (e.g., `"8Gi"`, `"16Gi"`)
  - `disk`: Disk space (e.g., `"50Gi"`, `"100Gi"`)
- Falls back to `package.resources` if not specified

### CLI Enhancements
The `melange test` command now supports resource override flags:
- `--cpu` - Override CPU allocation
- `--cpumodel` - Specify CPU model requirements  
- `--memory` - Override memory allocation
- `--disk` - Override disk space
- `--timeout` - Override test timeout

### Resource Precedence

**For external schedulers:**
1. `test-resources` (if specified in config)
2. `resources` (fallback)

**For local testing with `melange test`:**
1. CLI flags (highest priority)
2. `test-resources` from config
3. `resources` from config

### 1. Build-Heavy, Test-Light
```yaml
package:
  name: gcc-13
  resources:
    cpu: "8"
    memory: "16Gi"
  test-resources:
    cpu: "4"
    memory: "8Gi"
```
Compilation needs significant resources, but unit tests can run with less.

### 2. Test-Heavy
```yaml
package:
  name: tensorflow-lite
  resources:
    cpu: "4"
    memory: "8Gi"
  test-resources:
    cpu: "32"
    memory: "128Gi"
```
Simple build, but integration tests with ML models need substantial resources.

### 3. Test-Only Resources
```yaml
package:
  name: nginx-module-geoip
  test-resources:
    cpu: "4"
    memory: "8Gi"
```
No specific build resources needed, but tests require certain allocations.

## Testing

- ✅ 9 test cases covering various configurations and edge cases
- ✅ All existing tests pass
- ✅ Pre-commit hooks pass (go fmt, go ineffassign, shellcheck)

## Documentation

- ✅ Updated `docs/BUILD-FILE.md` with comprehensive resource documentation
- ✅ Updated `docs/md/melange_test.md` with new CLI flags
- ✅ Added 3 realistic example YAML files
- ✅ Updated JSON and CUE schemas

## Backward Compatibility

Fully backward compatible:
- Existing configs without `test-resources` continue to work unchanged
- External schedulers can fall back to `resources` field
- No breaking changes to existing behavior

### CODE REVIEW

File: pkg/config/config_test.go [hunk 1]
```diff
@@ -1387,3 +1387,266 @@ func TestPackageURL(t *testing.T) {
...
+				CPUModel: "intel-xeon",
+				Memory:   "8Gi",
+			},
+			expectParseError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fp := filepath.Join(os.TempDir(), "melange-test-resources-"+tt.name)
```
Comment (sergiodj) [inline pkg/config/config_test.go]: If you use `t.TempDir()` here, you don't need to worry about `defer`ing the file deletion afterwards.

File: pkg/config/config_test.go [hunk 2]
```diff
@@ -1387,3 +1387,266 @@ func TestPackageURL(t *testing.T) {
...
+				require.Equal(t, "intel-xeon-platinum", cfg.Package.TestResources.CPUModel)
+				require.Equal(t, "256Gi", cfg.Package.TestResources.Memory)
+				require.Equal(t, "2Ti", cfg.Package.TestResources.Disk)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fp := filepath.Join(os.TempDir(), "melange-test-resources-edge-"+tt.name)
```
Comment (sergiodj) [inline pkg/config/config_test.go]: Likewise.

**Review by sergiodj — APPROVED** [review]
Thanks, @javacruft .  I like the simplicity of the design here.  I read through the code and it LGTM.  I left two small comments but they can be addressed later (if at all).
